### ISSUE TITLE
(mkvtoolnix ) package fails to install with error "Cannot process argument transformation on parameter 'file64'"

### ISSUE THREAD

**Issue by superbonaci** [chocolatey-community/chocolatey-coreteampackages#1688]
## Expected Behavior

`mkvtoolnix` package should upgrade from version v58.0.0 to v59.0.0

## Current Behavior

Fails to upgrade `mkvtoolnix` package.

## Possible Solution

No idea.

## Steps to Reproduce (for bugs)

1. Run the upgrade command on PowerShell. This is the log:
````
choco upgrade mkvtoolnix
Chocolatey v0.10.15
Upgrading the following packages:
mkvtoolnix
By upgrading you accept licenses for the packages.

You have mkvtoolnix v58.0.0 installed. Version 59.0.0 is available based on your source(s).
Progress: Downloading mkvtoolnix 59.0.0... 100%

mkvtoolnix v59.0.0 [Approved]
mkvtoolnix package files upgrade completed. Performing other installation steps.
ERROR: Cannot process argument transformation on parameter 'file64'. Cannot convert value to type System.String.
The upgrade of mkvtoolnix was NOT successful.
Error while running 'C:\ProgramData\chocolatey\lib\mkvtoolnix\tools\chocolateyInstall.ps1'.
 See log for details.

Chocolatey upgraded 0/1 packages. 1 packages failed.
 See the log for details (C:\ProgramData\chocolatey\logs\chocolatey.log).

Failures
 - mkvtoolnix (exited -1) - Error while running 'C:\ProgramData\chocolatey\lib\mkvtoolnix\tools\chocolateyInstall.ps1'.
 See log for details.

Did you know the proceeds of Pro (and some proceeds from other
 licensed editions) go into bettering the community infrastructure?
 Your support ensures an active community, keeps Chocolatey tip top,
 plus it nets you some awesome features!
 https://chocolatey.org/compare
````

## Your Environment

* Package Version used: v59.0.0
* Operating System and version: W10 21H1
* Chocolatey version: 0.10.15
* Install/uninstall gist: no idea what it is

**Comment by pauby** [comment]
@superbonaci Thanks for the issue. Can you go back and submit the issue using the provided template?

**Comment by superbonaci** [comment]
@pauby first post updated

**Comment by pauby** [comment]
I've just performed an install of `mkvtoolnix` v58.0

```
PS C:\Users\WDAGUtilityAccount> choco install mkvtoolnix --version 58.0
Chocolatey v0.10.15
Installing the following packages:
mkvtoolnix
By installing you accept licenses for the packages.
Progress: Downloading mkvtoolnix 58.0.0... 100%

chocolatey-core.extension v1.3.5.1
chocolatey-core.extension package files install completed. Performing other installation steps.
 Installed/updated chocolatey-core extensions.
 The install of chocolatey-core.extension was successful.
  Software installed to 'C:\ProgramData\chocolatey\extensions\chocolatey-core'

mkvtoolnix v58.0.0 [Approved]
mkvtoolnix package files install completed. Performing other installation steps.
Installing 64-bit mkvtoolnix...
mkvtoolnix has been installed.
mkvtoolnix installed to 'C:\Program Files\MKVToolNix'
Adding to PATH if needed
PATH environment variable does not have C:\Program Files\MKVToolNix in it. Adding...
  mkvtoolnix may be able to be automatically uninstalled.
Environment Vars (like PATH) have changed. Close/reopen your shell to
 see the changes (or in powershell/cmd.exe just type `refreshenv`).
 The install of mkvtoolnix was successful.
  Software installed as 'exe', install location is likely default.

Chocolatey installed 2/2 packages.
 See the log for details (C:\ProgramData\chocolatey\logs\chocolatey.log).
```

And then an upgrade

```
PS C:\Users\WDAGUtilityAccount> choco upgrade mkvtoolnix
Chocolatey v0.10.15
Upgrading the following packages:
mkvtoolnix
By upgrading you accept licenses for the packages.

You have mkvtoolnix v58.0.0 installed. Version 59.0.0 is available based on your source(s).
Progress: Downloading mkvtoolnix 59.0.0... 100%

mkvtoolnix v59.0.0 [Approved]
mkvtoolnix package files upgrade completed. Performing other installation steps.
Installing 64-bit mkvtoolnix...
mkvtoolnix has been installed.
mkvtoolnix installed to 'C:\Program Files\MKVToolNix'
Adding to PATH if needed
  mkvtoolnix may be able to be automatically uninstalled.
 The upgrade of mkvtoolnix was successful.
  Software installed as 'exe', install location is likely default.

Chocolatey upgraded 1/1 packages.
 See the log for details (C:\ProgramData\chocolatey\logs\chocolatey.log).
```

There were no issues on a new Windows 10 Sandbox machine.

> Install/uninstall gist: no idea what it is

A gist is a place where you can provide the logs and link to it from the issue (it saves you adding the logs to the issue which just overwhelms the issue with data and makes it more difficult to find what is needed. See [Creating gists](https://docs.github.com/en/github/writing-on-github/editing-and-sharing-content-with-gists/creating-gists) to create a public gist and update the first comment.

**Comment by superbonaci** [comment]
Here's my gist from `C:\ProgramData\chocolatey\logs\chocolatey.log`:
https://gist.github.com/superbonaci/58a4f8bddda8429152a4081c053967f5

**Comment by superbonaci** [comment]
The issue is still present. Is this package abandoned?

**Comment by superbonaci** [comment]
Finally could fix it myself performing a new installation:
````
choco uninstall mkvtoolnix gmkvextractgui
choco-cleaner.bat
choco install mkvtoolnix gmkvextractgui
````

**Comment by AdmiringWorm** [comment]
By looking at the log you provided, it seems there was 2 64bit executables available in the package tools directory.

Most likely, this happened to a previous installation/upgrade of the package was not able to remove the executables after installation (most likely the installer had not unlocked its handle on the file).

This would also explain why uninstalling and installing the package again worked.

The relevant part of the log lists.
```
2021-07-14 11:04:50,377 17936 [DEBUG] -  Found 'C:\ProgramData\chocolatey\lib\mkvtoolnix\tools\mkvtoolnix-64-bit-58.0.0-setup.exe'
  with checksum '31B3AFE9FF42E01B1527B1C7A2A577B9'
2021-07-14 11:04:50,437 17936 [DEBUG] -  Found 'C:\ProgramData\chocolatey\lib\mkvtoolnix\tools\mkvtoolnix-64-bit-59.0.0-setup.exe'
  with checksum '4FF2A9C21FEB2C82BBD71ABE4DDE7F84'
```

Re-Opening the issue until we have resolved the problem (easiest solution will be to rely on exact file names in the package, instead of globbing).

**Comment by AdmiringWorm** [comment]
Fixes has been added for this issue, and a new fix version has been scheduled to be pushed to the community repository.

**Comment by superbonaci** [comment]
Did you find the issue or it's trial and error update?

**Comment by AdmiringWorm** [comment]
@superbonaci I mentioned in one of my previous comments what the issue was.

The issue was that there were multiple executables in the installation directory that would get picked up when we tried to install the application (due to us using a globbing pattern to discover the files).

The reason for the multiple executables has not been determined though, and I can only assume it was either not unlocked when we tried to remove it in a previous installation/upgrade, or it had been locked by something else (like an anti-virus scanner) when we tried to remove it.

While that file will still not be removed in such cases, there should now not be a problem when trying to upgrade from a previous version when there are multiple files since I switched away from using a globbing pattern.
